File: atv12/exerc12.go
package main
import "fmt"

type Produto struct {
    Nome  string
    Preco float64
}

func (p Produto) SomarPreco(outro Produto) float64 {
    return p.Preco + outro.Preco
}

func main() {
    p1 := Produto{"Produto A", 100.0}
    p2 := Produto{"Produto B", 200.0}

    fmt.Println("Soma dos preços: R$", p1.SomarPreco(p2))
}

File: atv12/exerc12.py
class Produto:
    def __init__(self, nome, preco):
        self.nome = nome
        self.preco = preco

    def __add__(self, outro):
        return self.preco + outro.preco

p1 = Produto("Produto A", 100.0)
p2 = Produto("Produto B", 200.0)
print(f"Soma dos preços: R${p1 + p2}")
